main/solution/backend/config/infra/cloudformation.yml

```diff
@@ -320,205 +469,162 @@ Resources:
+              - !GetAtt [WfAssignmentsDb, Arn]
+              - !Join ['', [!GetAtt [WfAssignmentsDb, Arn], '/index/*']]
+              - !GetAtt [EnvironmentsDb, Arn]
+              - !GetAtt [EnvironmentsScDb, Arn]
+              - !Join ['', [!GetAtt [EnvironmentsScDb, Arn], '/index/*']]
+              - !GetAtt [EnvironmentsTypesDb, Arn]
+              - !GetAtt [StudiesDb, Arn]
+              - !GetAtt [ProjectsDb, Arn]
+              - !GetAtt [StudyPermissionsDb, Arn]
+              - !GetAtt [IndexesDb, Arn]
+              - !GetAtt [ProjectsDb, Arn]
+              - !GetAtt [AccountsDb, Arn]
+              - !GetAtt [AwsAccountsDb, Arn]
+              - !GetAtt [AuthenticationProviderTypesDb, Arn]
+              - !GetAtt [AuthenticationProviderConfigsDb, Arn]
+              - !GetAtt [RevokedTokensDb, Arn]
+              - !Join ['', [!GetAtt [StudiesDb, Arn], '/index/*']]
+              - !GetAtt [IndexesDb, Arn]
+              - !GetAtt [CostApiCachesDb, Arn]
+              - !GetAtt [UserRolesDb, Arn]
+              - !GetAtt [StorageGatewayDb, Arn]
+              - !GetAtt [DsAccountsDb, Arn]
+              - !GetAtt [RoleAllocationsDb, Arn]
+              - !GetAtt [ResourceUsagesDb, Arn]
+          - Effect: Allow
+            Action:
+              - ssm:GetParameter
+            Resource:
+              - !Sub 'arn:aws:ssm:${AWS::Region}:${AWS::AccountId}:parameter/${self:custom.settings.paramStoreRoot}/*'
+              - !Sub 'arn:aws:ssm:${AWS::Region}:*:parameter/aws/service/storagegateway/ami/FILE_S3/latest'
+          - Effect: Allow
+            Action:
+              - ssm:PutParameter
+              - ssm:DeleteParameter
+            Resource:
+              - !Sub 'arn:aws:ssm:${AWS::Region}:${AWS::AccountId}:parameter/${self:custom.settings.paramStoreRoot}/*'
+          - Effect: 'Allow'
+            Action:
+              - route53:ChangeResourceRecordSets
+            Resource: 'arn:aws:route53:::hostedzone/${self:custom.settings.hostedZoneId}'
+          - Effect: Allow
+            Action:
+              - states:StartExecution
+            Resource:
+              - !Sub 'arn:aws:states:${AWS::Region}:${AWS::AccountId}:stateMachine:${self:custom.settings.workflowStateMachineName}'
+          - Effect: Allow
+            Action:
+              - ec2:StartInstances
+              - ec2:StopInstances
+            Resource: !Sub 'arn:aws:ec2:${AWS::Region}:*:instance/*'
+          - Effect: Allow
+            Action:
+              - ec2:DescribeSubnets
+              - ec2:DescribeVpcs
+              - ec2:DescribeNetworkInterfaces
+              - ec2:DescribeInstanceStatus
+              - ec2:DescribeInstances
+              - ec2:DescribeImages
+              - ec2:DescribeVolumes
+              - ec2:DescribeSpotPriceHistory
+            Resource: '*' # For the actions listed above IAM does not support resource-level permissions and requires all resources to be chosen
```

Comment:
Is this policy used by cross account as well? Curious why workflow runner which runs in main account needs these permissions. My understanding was that we should be assuming the cross account role and performing these actions so this policy doesn't need these actions.

Reply:
The Describe* actions are required during provisioning since the AMIs reside in the main account.

Reply:
Synced up on Slack: Actions related to networking and AMIs are required since a specific VPC and Subnet can be configured to use for creating AMIs. Other actions can be removed.